src/AccessLogs.sol

```diff
@@ -138,10 +145,48 @@ library AccessLogs {
         Memory.PhysicalAddress writeAddress,
         uint64 newValue
     ) internal pure {
-        writeLeaf(
-            a,
-            Memory.strideFromWordAddress(writeAddress),
-            keccak256(abi.encodePacked(solidityUint64ToMachineWord(newValue)))
+        (Memory.PhysicalAddress leafAddress, uint64 wordOffset) =
```

Comment:
@GCdePaula @stephenctw @diegonehab 
I think we are all on the same page as to how to implement `writeWord()`, right?
What do you think of this new version?

Reply:
I think it works. I'll using the current implementation nomenclature.

The issue before was that the `readData` wasn't. Currently, it kinda seems like it isn't being verified either. However, its hash is being verified against the following value `loggedReadHash` in the buffer (not consuming `loggedReadHash`).

Then, `loggedReadHash` is being verified to be in the tree. By transitivity, we know `readData` was in the tree.

Is my understanding correct?